sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -0,0 +1,170 @@
+        goes back to the UNIX start date (default: None)
+    booktype : str
+        Look for observations or operations data or both (default: both)
+    verbosity : int
+        Output verbosity. 0:Error, 1:Warning, 2:Info(default), 3:Debug
+
+    """
+    bookcart = []
+    bookcartobsdb = ObsDb()
+
+    if booktype=="both":
+        accept_type = ["obs", "oper"]
+    else:
+        accept_type = [booktype]
+
+    config_dict = yaml.safe_load(open(config, "r"))
+    try:
+        base_dir = config_dict["base_dir"]
+    except KeyError:
+        if verbosity==0:
+            print("No base directory base_dir specified in config file!")
+    if "obsdb" in config_dict:
+        if os.path.isfile(config_dict["obsdb"]):
+            bookcartobsdb = ObsDb.from_file(config_dict["obsdb"])
+    if "obsdb_cols" in config_dict:
+        col_list = []
+        for col, typ in config_dict["obsdb_cols"].items():
+            col_list.append(col+" "+typ)
+        bookcartobsdb.add_obs_columns(col_list)
+
+    #How far back we should look
+    tnow = time.time()
+    if recency is not None:
+        tback = tnow - recency*86400
+    else:
+        tback = 0 #Back to the UNIX Big Bang 
+
+    #Find folders that are book-like and recent
+    for dirpath,_, _ in os.walk(base_dir):
+        last_mod = max(os.path.getmtime(root) for root,_,_ in os.walk(dirpath))
+        if last_mod<tback:#Ignore older directories
+            continue
+        if os.path.exists(os.path.join(dirpath, "M_index.yaml")):
+            #Looks like a context file
+            bookcart.append(dirpath)
+    #Check the books for the observations we want
+
+    for bookpath in bookcart:
+        if check_meta_type(bookpath) in accept_type:
+            #obsfiledb creation
+            checkbook(bookpath, config, add=True, overwrite=True)
+
+            index = yaml.safe_load(open(os.path.join(bookpath, "M_index.yaml"), "rb"))
+            obs_id = index.pop("book_id")
+            tags = index.pop("tags")
+            detsets = index.pop("detsets")
+
+            if "obsdb_cols" in config_dict:
+                very_clean = {col:index[col] for col in iter(config_dict["obsdb_cols"]) if col in index}
+            else:
+                col_list = []
+                clean = {key:val for key, val in index.items() if val is not None}
+                very_clean = {key:val for key, val in clean.items() if type(val) is not list}
+                for key, val in very_clean.items():
+                    col_list.append(key+" "+type(val).__name__)
+                bookcartobsdb.add_obs_columns(col_list)
+
+            start = index.get("start_time")
+            end = index.get("end_time") 
+            if None not in [start, end]:
+                bookcartobsdb.add_obs_columns(["timestamp float", "duration float"])
+                very_clean["timestamp"] = start
+                very_clean["duration"] = end - start
+            
+            if tags != ([] or [""]):
```

Comment:
That's probably not what you want ... I think you want `if tags != [] and tags != [""]`.  Or perhaps `if (len([t for t in tags if t != '']) > 0`

Reply:
Good catch I'll use the `and` formulation.